tests/scripts/run-pmi-diffs.py

```diff
@@ -0,0 +1,704 @@
+
+    # Clone at that commit
+
+    command = 'git clone -b %s --single-branch %s %s' % (
+        target_branch, Coreclr_url, baseCoreClrPath)
+    log(command)
+    returncode = 0 if testing else os.system(command)
+    if returncode != 0:
+        log('ERROR: git clone failed')
+        return 1
+
+    # Change directory to the baseline root
+
+    cwd = os.getcwd()
+    log('[cd] %s' % baseCoreClrPath)
+    if not testing:
+        os.chdir(baseCoreClrPath)
+
+    # Set up for possible docker usage
+
+    scriptPath = '.'
+    buildOpts = ''
+    dockerCmd = ''
+    if not Is_windows and (arch == 'arm' or arch == 'arm64'):
+        # Linux arm and arm64 builds are cross-compilation builds using Docker.
+        if arch == 'arm':
+            dockerFile = 'microsoft/dotnet-buildtools-prereqs:ubuntu-14.04-cross-e435274-20180426002420'
+            dockerOpts = '-e ROOTFS_DIR=/crossrootfs/arm -e CAC_ROOTFS_DIR=/crossrootfs/x86'
+        else:
+            # arch == 'arm64'
+            dockerFile = 'microsoft/dotnet-buildtools-prereqs:ubuntu-16.04-cross-arm64-a3ae44b-20180315221921'
+            dockerOpts = '-e ROOTFS_DIR=/crossrootfs/arm64'
+
+        dockerCmd = 'docker run -i --rm -v %s:%s -w %s %s %s ' % (baseCoreClrPath, baseCoreClrPath, baseCoreClrPath, dockerOpts, dockerFile)
+        buildOpts = 'cross crosscomponent'
+        scriptPath = baseCoreClrPath
+
+    # Build a checked baseline jit 
+
+    if Is_windows:
+        command = 'set __TestIntermediateDir=int&&build.cmd %s checked skiptests skipbuildpackages' % arch
+    else:
+        command = '%s%s/build.sh %s checked skiptests skipbuildpackages %s' % (dockerCmd, scriptPath, arch, buildOpts)
+    log(command)
+    returncode = 0 if testing else os.system(command)
+    if returncode != 0:
+        log('ERROR: build failed')
+        return 1
+
+    # Build the layout (Core_Root) directory
+
+    # For Windows, you need to first do a restore. It's unfortunately complicated. Run:
+    #   run.cmd build -Project="tests\build.proj" -BuildOS=Windows_NT -BuildType=Checked -BuildArch=x64 -BatchRestorePackages
+
+    if Is_windows:
+        command = 'run.cmd build -Project="tests\\build.proj" -BuildOS=Windows_NT -BuildType=%s -BuildArch=%s -BatchRestorePackages' % (build_type, arch)
+        log(command)
+        returncode = 0 if testing else os.system(command)
+        if returncode != 0:
+            log('ERROR: restoring packages failed')
+            return 1
+
+    if Is_windows:
+        command = 'tests\\runtest.cmd %s checked GenerateLayoutOnly' % arch
+    else:
+        command = '%s%s/build-test.sh %s checked generatelayoutonly' % (dockerCmd, scriptPath, arch)
+    log(command)
+    returncode = 0 if testing else os.system(command)
+    if returncode != 0:
+        log('ERROR: generating layout failed')
+        return 1
+
+    # After baseline build, change directory back to where we started
+
+    log('[cd] %s' % cwd)
+    if not testing:
+        os.chdir(cwd)
```

Comment:
This is an unfortunate dependency. If there are any exceptions this will not be executed. The easiest way around this is to create a class that has an __enter__ and __exit__ method which will force the chdir to happen in the enter, and back in the exit. This would require adding a with statement to this method.